buku.py

```diff
@@ -4361,6 +4383,8 @@ def main():
             else:
                 # Search URLs, titles, tags for any keyword
                 search_results = bdb.searchdb(args.sany, False, args.deep)
+                if exclude_results:
```

Comment:
I think the order should be:

1. search keywords
2. 1 - search without keywords
3. 2 - search tags

Reply:
Not sure I understand the why order change.  The way its written now, search_tags is exclusive, as is search_keywords.  If we change the order, the conditional would be `if not tag_search: ...`, but the execution of the queries would be the same regardless of the order.

Reply:
With the current logic, if we apply:

`--sany a b c --stag d e f --exclude g h i`

the `exclude` part will never execute.

At least we should make `if exclude_results` independent i.e. pull it out of the `tags_search` case.

I will make the change.

I will make the change.

Reply:
Makes sense now, thanks.